src/hoc/withTheme.js

```diff
@@ -31,18 +32,21 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
     unsubscribeId: number = -1
 
     componentWillMount() {
+      const { defaultProps } = this.constructor
       const styledContext = this.context[CHANNEL_NEXT]
-      if (styledContext === undefined) {
+      const defaultTheme = determineTheme(this.props, undefined, defaultProps)
+      if (!styledContext && !defaultTheme) {
         // eslint-disable-next-line no-console
-        console.error('[withTheme] Please use ThemeProvider to be able to use withTheme')
-        return
+        console.warn('[withTheme] You are not using a ThemeProvider nor passing a theme prop or a theme in defaultProps')
+      } else if (!styledContext && defaultProps) {
```

Comment:
For this `else if` statement and the `if` statement above:

Can you please change it to explicit `!==` and `===` `undefined` please? This would save us some time when optimising this for v8 perf.